services/director-v2/src/simcore_service_director_v2/modules/dynamic_sidecar/scheduler/task.py

```diff
@@ -357,14 +357,23 @@ async def observing_single_service(service_name: str) -> None:
                 logger.debug("completed observation cycle of %s", f"{service_name=}")
             except asyncio.CancelledError:  # pylint: disable=try-except-raise
                 raise  # pragma: no cover
-            except Exception:  # pylint: disable=broad-except
+            except Exception as e:  # pylint: disable=broad-except
                 service_name = scheduler_data.service_name
 
-                message = (
-                    f"Observation of {service_name} failed:\n{traceback.format_exc()}"
+                # With unhandled errors, let's generate and ID and send it to the end-user
+                # so that we can trace the logs and debug the issue.
+
+                error_code = create_error_code(e)
+                logger.exception(
+                    "Observation of %s unexpectedly failed",
+                    f"{service_name=}",
+                    extra={"error_code": error_code},
+                )
+                scheduler_data.dynamic_sidecar.status.update_failing_status(
+                    # This message must be human-friendly
+                    f"Upss! This service ({service_name}) unexpectedly failed",
```

Comment:
Let's go with the American version?
```suggestion
                    f"Ops! This service ({service_name}) unexpectedly failed",
```